src/vs/workbench/browser/parts/editor/multiEditorTabsControl.ts

```diff
@@ -684,7 +684,20 @@ export class MultiEditorTabsControl extends EditorTabsControl {
 		this.layout(this.dimensions, { forceRevealActiveTab: true });
 	}
 
+	private previousSelectedCount = 0;
 	updateEditorSelections(): void {
+		// We only need to redraw from here when a selection got removed
+		// otherwise it will be handled by the open editor change event.
+		// Checking count is currently enough but might require checking
+		// each editor in the future if selection is changed programatically.
+		const newSelectedCount = this.tabsModel.selectedEditors.length;
```

Comment:
I wonder if this assumption should be made by the component that calls this method because that is the one knowing more about open editors behaviour.